extra_data/reader.py

```diff
@@ -581,17 +586,340 @@ def union(self, *others):
         sources_data = {src: src_datas[0].union(*src_datas[1:])
                         for src, src_datas in sources_data_multi.items()}
 
+        # Check for alias conflicts across all data collections.
+        duplicate_aliases = set(self._aliases.keys()).intersection(
+            *[dc._aliases for dc in others])
+
+        for alias in duplicate_aliases:
+            # Find all values occuring for duplicate aliases.
+            duplicate_alias_values = set([self._aliases[alias]]).union(
+                [dc._aliases[alias] for dc in others])
+
+            if len(duplicate_alias_values) > 1:
+                raise ValueError('conflicting aliases across data collections '
+                                 'to be merged')
+
+        aliases = self._aliases.copy()
+        for dc in others:
+            aliases.update(dc._aliases)
+
         train_ids = sorted(set().union(*[sd.train_ids for sd in sources_data.values()]))
         files = set().union(*[sd.files for sd in sources_data.values()])
 
         return DataCollection(
             files, sources_data=sources_data, train_ids=train_ids,
-            inc_suspect_trains=self.inc_suspect_trains,
+            aliases=aliases, inc_suspect_trains=self.inc_suspect_trains,
             is_single_run=same_run(self, *others),
         )
 
-    def _expand_selection(self, selection):
+    def _parse_aliases(self, alias_defs):
+        aliases = {}
+
+        def is_valid_alias(k, v):
+            return (isinstance(k, str) and (
+                isinstance(v, str) or (isinstance(v, tuple) and len(v) == 2)
+            ))
+
+        for alias_def in alias_defs:
+            if isinstance(alias_def, Mapping):
+                if any([not is_valid_alias(k, v) for k, v in alias_def.items()]):
+                    raise ValueError('alias definition by dict must be all '
+                                     'str keys to str values for sources or '
+                                     '2-len tuples for sourcekeys')
+
+                aliases.update(alias_def)
+            elif isinstance(alias_def, (str, os.PathLike)):
+                # From a file.
+                aliases.update(self._load_aliases_from_file(Path(alias_def)))
+
+        return aliases
+
+    def _load_aliases_from_file(self, aliases_path):
+        """Load alias definitions from file."""
+
+        if aliases_path.suffix == '.json':
+            import json
+
+            with open(aliases_path, 'r') as f:
+                data = json.load(f)
+
+        elif aliases_path.suffix == '.yaml':
+            import yaml
+
+            with open(aliases_path, 'r') as f:
+                data = yaml.safe_load(f)
+
+        elif aliases_path.suffix == '.toml':
+            from sys import version_info as py_version
+
+            # Use the standard library's tomllib if possible and try
+            # tomli if necessary.
+            if py_version.major >= 3 and py_version.minor >= 11:
```

Comment:
I'd normally just try importing `tomllib` and fall back on ImportError, though I don't think it matters.